charts/reporting-operator/templates/reporting-operator-deployment.yaml

```diff
@@ -156,12 +156,21 @@ spec:
               name: reporting-operator-config
               key: prometheus-datasource-import-from
               optional: true
+{{- /* require tokenSecret XOR serviceAccount */ }}
+{{- required (.Values.spec.config.prometheusImporter.auth.tokenSecret.value || .Values.spec.config.prometheusImporter.auth.useServiceAccountToken) && !(.Values.spec.config.prometheusImporter.auth.tokenSecret.value || .Values.spec.config.prometheusImporter.auth.useServiceAccountToken)  }}
```

Comment:
You probably want this in separate `if` conditions so you can set an error message based on which case your hitting. Currently you have no error message for `required`.